api/src/utils/itis-sort.ts

```diff
@@ -0,0 +1,141 @@
+import { TaxonSearchResult } from '../services/taxonomy-service';
+
+/**
+ * Sorts the ITIS response by how strongly records match the search terms
+ *
+ * @param {ItisSolrSearchResponse[]} data
+ * @param {string[]} searchTerms
+ * @memberof ItisService
+ */
+export const sortExactMatches = (data: TaxonSearchResult[], searchTerms: string[]): TaxonSearchResult[] => {
+  const searchTermsLower = searchTerms.map((item) => item.toLowerCase());
+
+  // Prioritize taxa where any word in the scientific or common name matches ANY of the search terms
+  // eg. ['Black', 'bear'] -> "Black" matches on "Black widow"
+  const containsAnyMatch = customSortContainsAnyMatchingSearchTerm(data, searchTermsLower);
+
+  // Prioritize taxa where either the scientific name or any common name CONTAINS the search terms joined
+  // eg. ['Black', 'bear'] -> "Black bear" matches on "American black bear"
+  const containsAnyMatchJoined = customSortContainsSearchTermsJoinedExact(containsAnyMatch, searchTermsLower);
+
+  // Prioritize taxa where either the scientific name or any common name is EXACTLY EQUAL to the search terms joined
+  // eg. ['Wolf'] -> "Wolf" is prioritized over "Forest Wolf"
+  const exactlyEquals = customSortEqualsSearchTermsExact(containsAnyMatchJoined, searchTermsLower);
+
+  return exactlyEquals;
+};
+
+/**
+ * Sorts the ITIS response to prioritize records where any word in the scientific or
+ * common name matches ANY of the search terms
+ *
+ * @param {ItisSolrSearchResponse[]} data
+ * @param {string[]} searchTerms
+ * @memberof ItisService
+ */
+export const customSortContainsAnyMatchingSearchTerm = (
+  data: TaxonSearchResult[],
+  searchTerms: string[]
+): TaxonSearchResult[] => {
+  // Custom sorting function
+  const customSort = (a: TaxonSearchResult, b: TaxonSearchResult) => {
+    const aInReference = checkForMatch(a, searchTerms);
+    const bInReference = checkForMatch(b, searchTerms);
+
+    if (aInReference && !bInReference) {
+      return -1; // Place items from searchTerms before other items
+    } else if (!aInReference && bInReference) {
+      return 1; // Place other items after items from searchTerms
+    } else {
+      return 0; // Maintain the original order if both are from searchTerms or both are not
+    }
+  };
+
+  const checkForMatch = (item: TaxonSearchResult, searchTerms: string[]) => {
+    const commonNameWords = item.commonNames?.flatMap((name: string) => name.toLowerCase().split(/\s+/));
```

Comment:
`.split(/\s+/)` will yield an array that could contain an emptystring, (for example, `" hello world".split(/\s+/)` will yield `["", "hello", "world"]`) so you may want to filter these out (you can just add `.filter(Boolean)`). To that end, if you're filtering out empty strings, you may as well just split on `' '` rather than a Regex expression